jobs/nr-day-job/nr_day_job.py

```diff
@@ -0,0 +1,128 @@
+# Copyright © 2021 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""s2i based launch script to run the service."""
+import asyncio
+import os
+
+from flask import Flask, current_app
+from namex.models import Request, State, db
+from namex.services.queue import QueueService
+from sqlalchemy import text
+
+import config
+from utils.logging import setup_logging
+
+APP_CONFIG = config.get_named_config(os.getenv('FLASK_ENV', 'production'))
+
+
+async def run():
+    """Run the day job."""
+    app = Flask(__name__)
+    app.config.from_object(APP_CONFIG)
+    db.init_app(app)
+
+    register_shellcontext(app)
+
+    qsm = QueueService(app=app, loop=event_loop)
+
+    await notify_nr_before_expiry(app, qsm)
+    await notify_nr_expired(app, qsm)
+
+
+def register_shellcontext(app):
+    """Register shell context objects."""
+    def shell_context():
+        """Shell context objects."""
+        return {'app': app}
+
+    app.shell_context_processor(shell_context)
+
+
+async def publish_email_message(qsm: QueueService, payload: dict):  # pylint: disable=redefined-outer-name
+    """Publish the email message onto the NATS emailer subject."""
+    subject = APP_CONFIG.NATS_EMAILER_SUBJECT
+    current_app.logger.debug('publish to queue, subject:%s, event:%s', subject, payload)
+    await qsm.publish_json_to_subject(payload, subject)
+
+
+async def furnish_request_message(
+        qsm: QueueService,
+        request: Request,
+        option: str
+):  # pylint: disable=redefined-outer-name
+    """Send notification info to the mail queue."""
+    current_app.logger.debug('Start of the furnishing of request for %s nrNumber=%s', option, request.nrNum)
+    payload = {
```

Comment:
Unless there's a reason not to, we should leverage this payload style
https://github.com/bcgov/namex/blob/main/services/common/src/queue_common/messages.py#L35

Reply:
I used this style to align with entity.emailer https://github.com/bcgov/lear/blob/main/queue_services/entity-emailer/src/entity_emailer/worker.py#L107

Reply:
Changed in entity emailer and here